week4_2/src/pages/signup.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import apiClient from "../api/axios";
+import Contentbox from "../components/contentBox";
+import Title from "../components/Title";
+import InputForm from "../components/inputForm";
+import Cta from "../components/cta";
+
+const Signup = () => {
+  const [username, setUsername] = useState("");
+  const [nickname, setNickname] = useState("");
+  const [password, setPassword] = useState("");
+  const [passwordCheck, setPasswordCheck] = useState("");
+  const [isExist, setIsExist] = useState("");
+  const [signupButton, setSignupButton] = useState(false);
+  const [isClicked, setIsClicked] = useState(false);
+
+  const navigate = useNavigate();
+  const goToLogin = () => {
+    navigate(`/login`);
+  };
+
+  // 미입력시 회원가입 비활성화
+  const activateBtn = async () => {
+    if (
+      isClicked &&
+      username &&
+      nickname &&
+      password &&
+      passwordCheck &&
+      password === passwordCheck
+    ) {
+      await doubleCheck();
+      setSignupButton(true);
+      return true;
+    } else {
+      setSignupButton(false);
+      return false;
+    }
+  };
+
+  // 클릭시 이동 및 데이터 전송
+  const goToLoginHandler = async () => {
+    const ActivateSuccess = await activateBtn();
+    if (ActivateSuccess) {
+      try {
+        await postData();
+        goToLogin();
+        alert("회원가입 완료");
+      } catch (err) {
+        console.log("회원가입 실패", err);
+      }
+    } else {
+      console.log(ActivateSuccess);
+    }
+  };
+
+  // 회원정보 등록 API
+  const postData = async () => {
+    try {
+      const response = await apiClient.post("/api/v1/members", {
+        username: username,
+        nickname: nickname,
+        password: password,
+      });
```

Comment:
```suggestion
      const response = await apiClient.post("/api/v1/members", {
        username,
        nickname,
        password,
      });
```

오오 여기서 이렇게 쓸 수 있겠다 ㅎㅎ